### ISSUE TITLE
`fit` does not fail when passed inconsistent arguments

### ISSUE THREAD

**Issue by Puzzled-Face** [Roche/crmPack#733]
Consider

```
d <- .DefaultData()
m <- .DefaultLogisticLogNormalOrdinal()
o <- .DefaultMcmcOptions()
s <- mcmc(d, m, o)
fit(s, m, d, grade = 2L)
```

This should fail, because a `Data` object is inconsistent with a `LogisiticLogNormalOrdinal` model: because the concept of a grade 2 Toxicity is not defined for binary toxicity data.  But the code runs without complaint:

```
   dose       middle        lower       upper
1     1 0.0002537003 5.769897e-37 0.002000647
2     3 0.0004881760 1.419004e-27 0.004139096
3     5 0.0007034469 2.422936e-23 0.005833511
4    10 0.0012700314 1.902935e-17 0.010788006
5    15 0.0019338823 8.135069e-14 0.016612540
6    20 0.0027419927 2.343198e-11 0.022942224
7    25 0.0037475733 1.894745e-09 0.031614266
8    40 0.0089582431 2.622198e-05 0.065084805
9    50 0.0175901435 1.997202e-04 0.112505658
10   80 0.1400388242 6.455298e-04 0.989460143
11  100 0.2153154660 9.069558e-04 0.999857700
```

It's not clear what the results represent.

Questions:
-  What's the most efficient way to prevent this?
-  With the growing number of models and data types, do similar problems exist elsewhere?
    -  If so, how do we test for the existence of such issues and then - in the future - confirm that the problem has been resolved?

**Comment by Puzzled-Face** [comment]
In this particular case, 

```
setMethod(
  "fit",
  signature = signature(
    object = "Samples",
    model = "LogisticLogNormalOrdinal",
    data = "Data"
  ),
  def = function(object,
                 model,
                 data,
                 points = data@doseGrid,
                 quantiles = c(0.025, 0.975),
                 middle = mean,
                 ...) {
    stop(
      paste0(
        "The LogisticLogNormalOrdinal model cannot be used with a '",
        class(data),
        "' object.  Please supply observed data in a `DataOrdinal` object instead."
      )
    )
  }
)  
````

appears to solve the problem.

``` 
fit(s, m, d, grade = 2L)

Error in .local(object, model, data, ...) :
The LogisticLogNormalOrdinal model cannot be used with a 'Data' object. Please supply observed data in a `DataOrdinal` object instead.
```

**Comment by Puzzled-Face** [comment]
Fixed in #736
